File: src/main/java/com/neo/caption/ocr/config/JoinerBean.java
package com.neo.caption.ocr.config;

import com.google.common.base.Joiner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JoinerBean {

    @Bean(name = "dotJoiner")
    public Joiner dotJoiner() {
        return Joiner.on(".");
    }

    @Bean(name = "plusJoiner")
    public Joiner plusJoiner() {
        return Joiner.on("+");
    }

    @Bean(name = "arrowJoiner")
    public Joiner arrowJoiner() {
        return Joiner.on(" -> ");
    }

}

File: src/main/kotlin/com/neo/caption/ocr/config/JoinerBean.kt
package com.neo.caption.ocr.config

import com.google.common.base.Joiner
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class JoinerBean {

    @Bean
    fun dotJoiner() = Joiner.on(".")

    @Bean
    fun plusJoiner() = Joiner.on("+")

    @Bean
    fun arrowJoiner() = Joiner.on(" -> ")

}
